remsfal-services/remsfal-platform/src/main/java/de/remsfal/service/control/TenancyController.java

```
@@ -39,20 +50,67 @@ public TenancyEntity getTenancy(final UserModel tenant, final UUID tenancyId) {
             .orElseThrow(() -> new NotFoundException("Tenancy not exist"));
     }
 
-    @Transactional(TxType.MANDATORY)
-    public TenancyEntity updateTenancy(final UUID projectId, TenancyEntity entity, final TenancyModel tenancy) {
-        if(entity == null) {
-            entity = new TenancyEntity();
-            entity.generateId();
-            entity.setProjectId(projectId);
+    public List<TenancyEntity> getTenanciesByProject(final UUID projectId) {
+        logger.infov("Retrieving all tenancies (projectId = {0})", projectId);
+        return tenancyRepository.findTenancyByProject(projectId);
+    }
+
+    public TenancyEntity getTenancyByProject(final UUID projectId, final UUID tenancyId) {
+        logger.infov("Retrieving a tenancy (projectId = {0}, tenancyId = {1})", projectId, tenancyId);
+        return tenancyRepository.findTenancyByProject(projectId, tenancyId)
+            .orElseThrow(() -> new NotFoundException("Tenancy not exist"));
+    }
+
+    @Transactional
+    public TenancyEntity createTenancy(final UUID projectId, final TenancyModel tenancy) {
+        logger.infov("Creating a tenancy (project={0}", projectId);
+
+        if (projectRepository.findById(projectId) == null) {
+            throw new NotFoundException("Project not exist");
         }
-        if(tenancy.getStartOfRental() != null) {
+
+        TenancyEntity entity = new TenancyEntity();
+        entity.generateId();
+        entity.setProjectId(projectId);
+        entity.setStartOfRental(tenancy.getStartOfRental());
+        entity.setEndOfRental(tenancy.getEndOfRental());
+
+        final List<? extends CustomerModel> tenants = tenancy.getTenants();
+        if (tenants != null && !tenants.isEmpty()) {
+            List<UserEntity> userEntities = tenants.stream()
+                .map(user -> userRepository.findById(user.getId()))
+                .filter(Objects::nonNull)
```

Currently in `createTenancy` and `updateTenancy`, any user ids that do not exist in the database are silently ignored. This means that if an invalid or non-existent user id is passed, it will simply be skipped without any feedback. It might be worth considering explicit validation or throwing an exception when a user id is not found.